From: Thomas Guillem <thomas@gllm.fr>
Date: Wed, 11 Sep 2024 08:15:52 +0200
Subject: dts_header: initialize SUBSTREAM_LBR size to 0

It's a substream and can't work alone, so a size of 0 is legitimate.

Fixes #28773 (Use of uninitialised value)

(cherry picked from commit b23f09b258dc63c2221af502ec856630f68817e5)
Signed-off-by: Thomas Guillem <thomas@gllm.fr>
---
 modules/packetizer/dts_header.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/modules/packetizer/dts_header.c b/modules/packetizer/dts_header.c
index 53249dd..8d6d5c8 100644
--- a/modules/packetizer/dts_header.c
+++ b/modules/packetizer/dts_header.c
@@ -342,6 +342,7 @@ static int dts_header_ParseLBRExtSubstream( vlc_dts_header_t *p_header,
     bs_skip( &s, 16 );
     uint16_t nLBRScaledBitRate_LSW = bs_read( &s, 16 );
     p_header->i_bitrate = nLBRScaledBitRate_LSW | ((nLBRBitRateMSnybbles & 0xF0) << 12);
+    p_header->i_frame_size = 0;
     return VLC_SUCCESS;
 }
 
